Storage/src/StorageObject.php

```diff
@@ -672,13 +676,59 @@ public function downloadToFile($path, array $options = [])
      */
     public function downloadAsStream(array $options = [])
     {
-        return $this->connection->downloadObject(
+        $headers = $options['restOptions']['headers'] ?? [];
+        $resultStream = Utils::streamFor(null);
+        $userSuppliedRangeHeaders = isset($headers['Range']);
+        $expectedSize = isset($options['size']) ? $options['size'] : $this->info()['size'];
```

Comment:
If we do go this route, let's make sure to document providing size across this and any other methods using this underlying implementation.

I'm generally a bit wary about this, because if size isn't provided then a potentially unexpected network request will be triggered (the storage object could be lazily instantiated, e.g. `$client->bucket('bucket')->object('object')->downloadAsStream()`). I think I'd feel a little more comfortable if we did one of the following:

a) Put this resumable logic into another method, like`resumableDownloadAsStream` where it is made explicitly clear additional network requests might be initiated.
b) Add a flag, like `resumable`, which defaults to false and explcitly states the risk of additional network requests in order to preserve existing behavior.

Happy to discuss more and weigh trade offs, let me know :).

Reply:
After talking to Chris, I was able to deduce that we can infer that an object was a transcoded object without making a metadata call. Currently, we're using the `x-guploader-response-body-transformations` headers in the response, but I am verifying this from the Storage team. Nevertheless, the `info` call has been removed. Thanks for this callout :)

Reply:
We updated the header to be used to `X-Goog-Stored-Content-Encoding` after a discussion with the Storage team.